api/v1alpha1/mirrorpeer_types.go

```diff
@@ -46,6 +48,11 @@ type PeerRef struct {
 
 // MirrorPeerSpec defines the desired state of MirrorPeer
 type MirrorPeerSpec struct {
+	// Type represents the mode of DR operation (sync or async)
+	// +kubebuilder:default=async
+	// +kubebuilder:validation:Enum=async;sync
+	Type DRType `json:"type"`
```

Comment:
I'm thinking this could be in Status instead and be determined internally using CephFSID from CephCluster.

Reply:
As discussed, This change cannot be as controller on hub cannot identify  sync/async till the agents are running and hence we need the DRType api